### PR TITLE
Fixed #24073 -- Deactivated translations when leave_locale_alone is False

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: docs/howto/custom-management-commands.txt [hunk 1]
```diff
@@ -145,13 +145,18 @@ default options such as :djadminopt:`--verbosity` and :djadminopt:`--traceback`.
...
+By default, the :meth:`BaseCommand.execute` method deactivates translations
+because some commands shipped with Django perform several tasks (for example,
+user-facing content rendering and database population) that require a
+project-neutral string language.
 
-If, for some reason, your custom management command needs to use a fixed locale
-different from 'en-us', you should manually activate and deactivate it in your
+.. versionchanged:: 1.8
+
+    In previous versions, Django used to force the "en-us" locale instead of
```
Comment (timgraham) [inline docs/howto/custom-management-commands.txt]: Django forced...

**Comment by MarkusH** [comment]
I vote for a deprecation of the current behavior over "just changing it".

**Comment by claudep** [comment]
Markus, could you elaborate your opinion? What do you fear?

**Comment by MarkusH** [comment]
Lets assume you have `IntegerField(ugettext_lazy('Menge'))` in your models and your default language is `de-de`. You also have translation files that have a msgid `"Menge"` and a msgstr `"Amount"` (translation from German to English).

From what I understand, forcing `en-us` for all management commands, `makemigrations` yields a migration file with `IntegerField(ugettext_lazy('Amount'))`. If we now just deactivate all translations, the output would be `IntegerField(ugettext_lazy('Menge'))`.

Bottom line, this change generates new migrations for everybody who doesn't have `en-us` as the source language for translations.

**Comment by claudep** [comment]
I didn't try your examples, but assuming they prove right, that still means that in this sort of setup, migrations are created each time you change some translations, which is also a "crazy" situation. Moreover, I cannot see which deprecation path we could take. We could add some note under backwards incompatibility changes, but I'm still not convinced about going a (possibly difficult) deprecation route.

**Comment by MarkusH** [comment]
Looks like I was right: https://github.com/MarkusH/django-issue-24073

I you don't think there is a reasonable way for a deprecation path (I can't think of one either right now) I'd like to have some comment in the "backwards incompatible changes" section of the release notes.

**Comment by claudep** [comment]
I moved the release note to the Backwards incompatible changes section and beefed it up slightly.

**Comment by timgraham** [comment]
Note looks good to me.
